### ISSUE TITLE
Performance issue

### ISSUE THREAD

**Issue by alexislefebvre** [liip/LiipTestFixturesBundle#12]
### Preconditions
Performance issue with version 1.0 of this bundle: https://github.com/liip/LiipFunctionalTestBundle/issues/516#issuecomment-498755808 compared to LiipFunctionalTestBundle 2.0.0-alpha16

### Steps to reproduce
.

### Expected result
.

### Actual result
Tests are about 2x slower.

**Comment by alexislefebvre** [comment]
@Simperfit found that `createSchema` is taking much more time.

**Comment by alexislefebvre** [comment]
TODO : use Blackfire to compare performance with https://github.com/liip/LiipFunctionalTestBundle/releases/tag/2.0.0-alpha16 and LiipFunctionalTestBundle 3.0 and LiipTestFixturesBundle 1.0.

**Comment by alexislefebvre** [comment]
On a personal project, tests are 4 times slower. :thinking:

**Comment by alexislefebvre** [comment]
TODO: compare when using dependency injection as in #18.

**Comment by magnetik** [comment]
Could this be because kernel & containers are not shared between the two bundle so when using both `loadFixtures()` and `makeClient()` two kernels are being built?

**Comment by alexislefebvre** [comment]
@magnetik to my knowledge, event if some code is shared between the 2 bundles, there will be only one instance of the test class so there will be only one kernel built?

**Comment by magnetik** [comment]
When using both bundles, the two kernel built are from :
* `\Liip\FunctionalTestBundle\Test\WebTestCase::makeClient` through `\Symfony\Bundle\FrameworkBundle\Test\WebTestCase::createClient`
* `\Liip\TestFixturesBundle\Test\FixturesTrait::loadFixtures` (via `\Liip\TestFixturesBundle\Test\FixturesTrait::getContainer`)

Am I missing something?

**Comment by biozshock** [comment]
I found out why in our project we had a slowdown of about 3x.

According to the https://github.com/liip/LiipTestFixturesBundle/blob/2e6e867da5bf3fcf5017583a521254b719c20934/src/DependencyInjection/Configuration.php#L49 setting you always recreate schema for the database https://github.com/liip/LiipTestFixturesBundle/blob/f97ccdfb0bc4ff386f084eb587e44cc104194c03/src/Services/DatabaseTools/ORMDatabaseTool.php#L129 .

In LiipFunctionalTestBundle v1.x you had that only when sqlite database was defined for tests https://github.com/liip/LiipFunctionalTestBundle/blob/1.x/Test/WebTestCase.php#L449 . 

In our case with the tests with MySQL (mysql, percona and mariadb) we had tests passing faster than it was with LiipFunctionalTestBundle 1.x

**Comment by alexislefebvre** [comment]
Thanks for the investigation @biozshock 

The first change to this code was in this PR: https://github.com/liip/LiipFunctionalTestBundle/pull/398/files#diff-3216cb8554ab63f3b7b8531ebcee5156L387

I'll continue to look at changes later.

**Comment by alexislefebvre** [comment]
We could add a note in the documentation that there are several options to create database and schema:

> If the schema doesn't change between tests:
> 1. define `keep_database_and_schema: true`
> 2. create database with `bin/console --env=test doctrine:database:create`:
> 3. create schema with `php app/console --env=test doctrine:schema:update --force` or `php app/console --env=test doctrine:migrations:migrate --no-interaction`
> 4. run PHPUnit
> 
> If the schema change between tests:
> 1. don't define `keep_database_and_schema: true`
> 2. run PHPUnit and database and schema will be created automatically, please note that this method is slower

**Comment by alexislefebvre** [comment]
> We could add a note in the documentation that there are several options to create database and schema:

Fixed in #120.

The performance issue should be fixed in the 2.x release.

**Comment by Dekrikh** [comment]
Performance with LiipTestFixturesBundle v2.6.0

I am experiencing significant performance issues with LiipTestFixturesBundle version 2.6.0 when using the loadFixtures method with zero insertions. Even with an empty set of fixtures as shown below, tests take 7 to 9 seconds to run, which seems abnormally high. **And without loadFixtures()** 260 ms

`$this->databaseTool->loadFixtures([ ExempleFixtures::class]);`

```php
 <?php

namespace App\DataFixtures;

use Doctrine\Bundle\FixturesBundle\Fixture;
use Doctrine\Persistence\ObjectManager;

class ExempleFixtures extends Fixture
{
    public function load(ObjectManager $manager)
    {

    }
}

```
